zebra-chain/src/value_balance.rs

```diff
@@ -0,0 +1,297 @@
+        ValueBalance {
+            transparent: transparent_amount,
+            ..Default::default()
+        }
+    }
+
+    /// Creates a [`ValueBalance`] from the given sprout amount.
+    pub fn from_sprout_amount(sprout_amount: Amount<C>) -> Self {
+        ValueBalance {
+            sprout: sprout_amount,
+            ..Default::default()
+        }
+    }
+
+    /// Creates a [`ValueBalance`] from the given sapling amount.
+    pub fn from_sapling_amount(sapling_amount: Amount<C>) -> Self {
+        ValueBalance {
+            sapling: sapling_amount,
+            ..Default::default()
+        }
+    }
+
+    /// Creates a [`ValueBalance`] from the given orchard amount.
+    pub fn from_orchard_amount(orchard_amount: Amount<C>) -> Self {
+        ValueBalance {
+            orchard: orchard_amount,
+            ..Default::default()
+        }
+    }
+
+    /// Get the transparent amount from the [`ValueBalance`].
+    pub fn transparent_amount(&self) -> Amount<C> {
+        self.transparent
+    }
+
+    /// Insert a transparent value balance into a given [`ValueBalance`]
+    /// leaving the other values untouched.
+    pub fn set_transparent_value_balance(
+        &mut self,
+        transparent_value_balance: ValueBalance<C>,
+    ) -> &Self {
+        self.transparent = transparent_value_balance.transparent;
+        self
+    }
+
+    /// Get the sprout amount from the [`ValueBalance`].
+    pub fn sprout_amount(&self) -> Amount<C> {
+        self.sprout
+    }
+
+    /// Insert a sprout value balance into a given [`ValueBalance`]
+    /// leaving the other values untouched.
+    pub fn set_sprout_value_balance(&mut self, sprout_value_balance: ValueBalance<C>) -> &Self {
+        self.sprout = sprout_value_balance.sprout;
+        self
+    }
+
+    /// Get the sapling amount from the [`ValueBalance`].
+    pub fn sapling_amount(&self) -> Amount<C> {
+        self.sapling
+    }
+
+    /// Insert a sapling value balance into a given [`ValueBalance`]
+    /// leaving the other values untouched.
+    pub fn set_sapling_value_balance(&mut self, sapling_value_balance: ValueBalance<C>) -> &Self {
+        self.sapling = sapling_value_balance.sapling;
+        self
+    }
+
+    /// Get the orchard amount from the [`ValueBalance`].
+    pub fn orchard_amount(&self) -> Amount<C> {
+        self.orchard
+    }
+
+    /// Insert an orchard value balance into a given [`ValueBalance`]
+    /// leaving the other values untouched.
+    pub fn set_orchard_value_balance(&mut self, orchard_value_balance: ValueBalance<C>) -> &Self {
+        self.orchard = orchard_value_balance.orchard;
+        self
+    }
+}
+
+impl<C> Default for ValueBalance<C>
+where
+    C: Constraint + Copy,
+{
+    fn default() -> Self {
+        let zero = Amount::zero();
+        Self {
+            transparent: zero,
+            sprout: zero,
+            sapling: zero,
+            orchard: zero,
+        }
+    }
+}
```

Comment:
Readability and consistency: Can we change `ValueBalance::default` into `ValueBalance::zero`?

It would match the `Amount` implementation, and it's clearer what the value actually is.

You can use `..ValueBalance::zero()` to fill structs just like `..Default::default()`.